src/os/bluestore/BlueFS.cc

```diff
@@ -3775,7 +4155,7 @@ int BlueFS::fsync(FileWriter *h)/*_WF_WD_WLD_WLNF_WNF*/
       }
     }
   }
-  if (old_dirty_seq) {
+  if (old_dirty_seq && !h->file->is_new_wal()) { // don't force flush on WAL
```

Comment:
May be we don't need fsync-ing after regular appends to WAL but what's about the cases when we've [pre]allocated some space for it? This should get into log and be fsynced.

Reply:
done with last commit, please take a look